codegen-sbt/src/main/scala/caliban/codegen/CompileTimeCalibanPlugin.scala

```diff
@@ -0,0 +1,450 @@
+package caliban.codegen
+
+import caliban.codegen.CalibanSourceGenerator.TrackedSettings
+import caliban.tools.compiletime.Utils
+import sbt.Keys._
+import sbt.io.IO.defaultCharset
+import sbt.{ Compile, Def, Project, _ }
+
+import java.io.File
+import java.nio.channels.FileChannel
+import java.nio.file.{ Path, StandardOpenOption }
+import scala.util.Using
+
+object CompileTimeCalibanServerPlugin extends AutoPlugin {
+  override def requires = plugins.JvmPlugin
+  override def trigger  = noTrigger
+
+  object autoImport extends caliban.tools.compiletime.Config {
+
+    /* ## Tasks and settings namespace
+     *
+     * (https://www.scala-sbt.org/1.x/docs/Plugins-Best-Practices.html#Using+a+%E2%80%9Cmain%E2%80%9D+task+scope+for+settings)
+     */
+    lazy val ctCalibanServer: TaskKey[Unit] = taskKey[Unit]("Plugin configuration keys namespace")
+
+    // ## Required Plugin configurations
+    lazy val ctCalibanServerSettings: SettingKey[Seq[(String, ClientGenerationSettings)]] =
+      settingKey[Seq[(String, ClientGenerationSettings)]](
+        "(Required) List of GraphQL API reference and client generation configuration"
+      )
+
+    // ## Plugin task
+    lazy val ctCalibanServerGenerate: TaskKey[Seq[File]] = taskKey[Seq[File]]("Internal task")
+  }
+  import autoImport._
+
+  private val helpMsg: String =
+    """
+      |Missing configuration for the "CompileTimeCalibanServerPlugin" plugin.
+      |
+      |You need to configure the `Compile / ctCalibanServer / ctCalibanServerSettings` setting.
+      |
+      |Here's an example of configuration:
+      |```
+      |lazy val server =
+      |  project
+      |    .in(file("modules/server"))
+      |    .enablePlugins(CompileTimeCalibanServerPlugin)
+      |    .settings(
+      |      Compile / ctCalibanServer / ctCalibanServerSettings ++=
+      |        Seq(
+      |          "io.example.reference.GraphQLApi.api" -> ClientGenerationSettings.default,
+      |          "io.example.reference.GraphQLApi.api" ->
+      |            ClientGenerationSettings(
+      |              packageName = "io.example.reference.generated",
+      |              clientName = "CalibanClient",
+      |            ),
+      |          "io.example.reference.GraphQLApi.api" ->
+      |            ClientGenerationSettings(
+      |              packageName = "io.example.reference.generated.split",
+      |              splitFiles = true
+      |            )
+      |        )
+      |    )
+      |```
+      |
+      |See documentation for more details: (TODO Add link)
```

Comment:
Can you add documentation for this new plugin? That would be in https://github.com/ghostdogpr/caliban/blob/master/vuepress/docs/docs/client.md

Reply:
That's the next step for me :)

Reply:
I'll open another PR for that documentation if that's ok for you?

Reply:
It's even better, so I can merge it after the next release.

Reply:
Let's do like this then :)